source/common/http/conn_pool_base.cc

```diff
@@ -94,27 +61,26 @@ void ConnPoolImplBase::tryCreateNewConnection() {
   }
 }
 
-void ConnPoolImplBase::attachRequestToClient(ActiveClient& client,
-                                             ResponseDecoder& response_decoder,
-                                             ConnectionPool::Callbacks& callbacks) {
-  ASSERT(client.state_ == ActiveClient::State::READY);
+void ConnPoolImplBase::attachRequestToClientImpl(Envoy::ConnectionPool::ActiveClient& client,
+                                                 void* context) {
+  ASSERT(client.state_ == Envoy::ConnectionPool::ActiveClient::State::READY);
 
   if (!host_->cluster().resourceManager(priority_).requests().canCreate()) {
     ENVOY_LOG(debug, "max requests overflow");
-    callbacks.onPoolFailure(ConnectionPool::PoolFailureReason::Overflow, absl::string_view(),
-                            nullptr);
+    onPoolFailure(client.real_host_description_, absl::string_view(),
+                  ConnectionPool::PoolFailureReason::Overflow, context);
     host_->cluster().stats().upstream_rq_pending_overflow_.inc();
   } else {
     ENVOY_CONN_LOG(debug, "creating stream", client);
-    RequestEncoder& new_encoder = client.newStreamEncoder(response_decoder);
 
     client.remaining_requests_--;
     if (client.remaining_requests_ == 0) {
       ENVOY_CONN_LOG(debug, "maximum requests per connection, DRAINING", client);
       host_->cluster().stats().upstream_cx_max_requests_.inc();
-      transitionActiveClientState(client, ActiveClient::State::DRAINING);
-    } else if (client.numActiveRequests() >= client.concurrent_request_limit_) {
-      transitionActiveClientState(client, ActiveClient::State::BUSY);
+      transitionActiveClientState(client, Envoy::ConnectionPool::ActiveClient::State::DRAINING);
```

Comment:
Should `Envoy::ConnectionPool` be renamed to `ConnectionPoolCommon` or similar? It would make it clearer which namespace is being referenced (reusing namespace names hurts readability in my opinion), and I think that code will always be abstract in some way.

Reply:
Yeah, I dislike same name - I thought it was an Envoyish convention but with two of us in agreement I'm happy to rename.
I'd be inclined to have 
ConnectionPoolBase
HttpConnectionPoolBase
Http1ConnectionPool
Http2ConnectionPool

where I'd do the first two in this PR, and the other two in a follow-up.  SGTY?

Reply:
I'm fine with that. I'd also be fine with leaving Http1::ConnectionPool and Http2::ConnectionPool how they are, because at least those two would not be using another namespace named ConnectionPool anymore. But for sure +1 on ConnectionPoolBase and HttpConnectionPoolBase.